### DISCUSSION TITLE
Image with 2 different color profiles

### DISCUSSION THREAD

**Discussion by alejandroLoz** [dlemstra/Magick.NET#1560] [Help] [answered]
So I have a "background" image that has an attached color profile:

![1f782a09-648c-4ab1-8901-a982438ef9d1--75a0fbc9-db63-4e62-b865-e621b8ff31ca](https://github.com/dlemstra/Magick.NET/assets/13033000/f4d51ec1-545e-4b1a-999d-fcf7e211f1ac)

color profile exif data:
![image](https://github.com/dlemstra/Magick.NET/assets/13033000/d3e80b22-e37d-492d-b1f0-d02380a0044d)

And another "clipart" image with a default sRGB color profile:

![body](https://github.com/dlemstra/Magick.NET/assets/13033000/5429fab8-04fa-4888-921e-0fbef2a7c256)

![image](https://github.com/dlemstra/Magick.NET/assets/13033000/feea5005-03fb-4f2d-a061-cd6b6cc29656)

When I combine them using the following code:

```
MagickImage background = new MagickImage("C:\\Users\\someUser\\Documents\\background.jpg");
MagickImage clipart = new MagickImage("C:\\Users\\someUser\\Documents\\clipart\\clipart.png");
clipart.Resize(1000,1000);
background.Composite(clipart);
background.Write("C:\\Users\\loski\\Documents\\ticket\\generated.jpg");
```

The clipart image ends up with what it looks like the inverted colors:

![generated](https://github.com/dlemstra/Magick.NET/assets/13033000/dd93dbc7-1cf2-4d3e-91db-2d512e9c843f)

What am I doing wrong? How can I fix it?

**Comment by dlemstra** [comment]
I don't have time to check this properly now but it looks like you are mixing CMYK and sRGB images.

**Reply by alejandroLoz** [reply]
Thank you so much @dlemstra! This will sound silly but I feel honored to receive your message. 😊
What you'd recommend/ what's the best practice to mix a CMYK with an sRGB image?

**Reply by dlemstra** [reply, marked as answer]
It all depends on what kind of output image you want to have. I suspect that you want to have an CMYK output image looking at your website. You should then change your sRGB image to CMYK with the `TransformColorSpace` method. You can find an example for that in documentation.

p.s. There is no need to feel honored as I am one of the few people that can answer your question 😃.

**Reply by alejandroLoz** [reply]
That worked! 
I'm attaching a snippet to use as a reference in case someone faces the same problem:

```
    IColorProfile sourceProfile = ColorProfile.SRGB;
    IColorProfile targetProfile = ColorProfile.USWebCoatedSWOP;
    if (baseImage.GetColorProfile() != null)
    {
        targetProfile = baseImage.GetColorProfile();
    }
    if (clipart.GetColorProfile() != null)
    {
        sourceProfile = clipart.GetColorProfile();
    }
    clipart.TransformColorSpace(sourceProfile, targetProfile);
```

**Reply by dlemstra** [reply]
Thanks for adding the snippet. And you don't need to check if the `sourceProfile` exists because method will only use it when the image has no color profile. But I just noticed that the check was not 100% correct so I pushed a patch to fix that in the next release. For now you can use `clipart.HasProfile("icc") || clipart.HasProfile("icm")` but after the next release you can remove that. And I would also advise you to only call `baseImage.GetColorProfile()` once because you would otherwise create an object twice.
